luaparse.js

```diff
@@ -907,15 +912,27 @@
           // Return it as is, without translating the byte.
           return '\\' + input.slice(sequenceStart, index);
         }
+        return raise({}, errors.hexadecimalDigitExpected, '\\' + input.slice(sequenceStart, index + 2));
+      case '\\':
+      case '\'':
+      case '"':
+      case '[':
+      case ']':
         return '\\' + input.charAt(index++);
       default:
         // \ddd, where ddd is a sequence of up to three decimal digits.
-        if (isDecDigit(input.charCodeAt(index))) {
-          while (isDecDigit(input.charCodeAt(++index)));
-          return '\\' + input.slice(sequenceStart, index);
+        while (isDecDigit(input.charCodeAt(index)) && index - sequenceStart < 3) index++;
+
+        if (index === sequenceStart && options.luaVersion >= 5.2) {
```

Comment:
I feel vaguely uneasy seeing a floating-point value being used for something that is essentially an enumeration.
```
$ jsc
>>> 5.1+0.1+0.1
5.299999999999999
```
To be fair, I doubt many people are going to actually try this. I guess floating-point numbers should work fine as enumerators, as long as you don't do arithmetic on them. Some memory may be wasted, but it's so small it's insignificant. And if you cared about memory usage, you wouldn't be using JavaScript in the first place.

I'm also not sure whether `>=` should be used here. After all, we don't know for sure how Lua 5.4 or 6.0 is going to behave; we may conjecture it's going to be similar to 5.3, but just as well it might need adjustments like 5.2 did. Lua developers aren't known for their commitment to backwards compatibility.

I'd use an integer value for `luaVersion` (`501`/`502`/`503`). Or even a string. And reject values for versions other than 5.1, 5.2 and 5.3.

But maybe it's just me being grumpy. I'd ask @oxyc what he thinks.